ArcadiaTeamsBot/Dialogs/NewRequestDialog.cs

```diff
@@ -1,63 +1,288 @@
 {
+    using System;
     using System.Collections.Generic;
+    using System.Linq;
     using System.Threading;
     using System.Threading.Tasks;
 
+    using AdaptiveCards;
+
+    using ArcadiaTeamsBot.CQRS.Abstractions;
+    using ArcadiaTeamsBot.CQRS.Abstractions.Commands;
+    using ArcadiaTeamsBot.ServiceDesk.Abstractions;
+    using ArcadiaTeamsBot.ServiceDesk.Abstractions.DTOs;
+    using ArcadiaTeamsBot.ServiceDesk.Requests.RequestType;
+    using ArcadiaTeamsBot.ServiceDesk.Requests.RequestTypeFactory;
+
+    using MediatR;
+
     using Microsoft.Bot.Builder;
     using Microsoft.Bot.Builder.Dialogs;
     using Microsoft.Bot.Schema;
 
-    public class NewRequestDialog : ComponentDialog
+    using Newtonsoft.Json.Linq;
+
+    internal class NewRequestDialog : ComponentDialog
     {
-        private const string Back = "Back";
+        private const string Submit = "Submit";
+        private const string Cancel = "Cancel";
+        private const string Yes = "Yes";
+        private const string No = "No";
+        private const string Button = "Button";
+        private const string Title = "Title";
+        private const string TypeId = "TypeId";
+        private const string InputValidation = "InputValidation";
+        private const string Priority = "Priority";
+        private const string Description = "Description";
+        private const string ExecutionDate = "Execution Date";
+        private const string Username = "ekaterina.kuznetsova@arcadia.spb.ru";
+        private readonly IRequestTypeUIFactory requestTypeUIFactory;
+        private readonly IMediator mediator;
+        private readonly IServiceDeskClient serviceDeskClient;
 
-        public NewRequestDialog() : base(nameof(NewRequestDialog))
+        public NewRequestDialog(IMediator mediator, IRequestTypeUIFactory requestTypeUIFactory, IServiceDeskClient serviceDeskClient) : base(nameof(NewRequestDialog))
         {
+            this.serviceDeskClient = serviceDeskClient;
+            this.mediator = mediator;
+            this.requestTypeUIFactory = requestTypeUIFactory;
+
             this.AddDialog(new WaterfallDialog(nameof(WaterfallDialog), new WaterfallStep[]
             {
-                InfoStep,
-                EndStep,
+                this.InputStep,
+                this.ChoiceStep,
+                this.EndStep
             }));
 
             this.AddDialog(new TextPrompt(nameof(TextPrompt)));
             this.InitialDialogId = nameof(WaterfallDialog);
+            this.AddDialog(new TextPrompt(InputValidation, this.ValidateForm));
         }
 
-        private static async Task<DialogTurnResult> InfoStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        private async Task<bool> ValidateForm(PromptValidatorContext<string> promptContext, CancellationToken cancellationToken)
         {
-            var attachments = new[]
+            var formData = (JObject)promptContext.Context.Activity.Value;
```

Comment:
You use same code thrice - to validate, in choice step and to get final values. It could be better to create internal class which will represent form data and return parsed form values from a method.